src/components/lib/ConnectWalletDialog.js

```diff
@@ -3,124 +3,197 @@ import Check from '../svgs/Check'
+				await wallet.connector.activate()
+				setAvailableAccounts(await wallet.connector.getAccounts())
+			} else {
+				await activateProvider(null, wallet)
+			}
+		} catch(error) {
+			setError(error.toString())
+		}
+	}
 
+	const activateProvider = async (selectedAccount, wallet = chosenWallet) => {
 		try {
+			if(wallet.isHardwareWallet) {
+				wallet.connector.setDefaultAccount(selectedAccount)
+			}
 			await activate(wallet.connector, undefined, true)
 			onConnected()
 		} catch(ex) {
 			setError(ex.toString())
-			throw ex
 		}
 	}
 
-	const ChooseWalletStep = () => {
-		return <>
-			<div className="title">Connect to a wallet</div>
-			<p>This wallet will be used to sign transactions on Ethereum.</p>
- 
-			<ul className='wallets'>
-				{
-					WALLETS.map(wallet => {
-						return <li className='wallet-option' onClick={() => chooseWallet(wallet)}>
-							<img src={wallet.icon} />
-							{wallet.showName && wallet.name}
-						</li>
-					})
-				}
-			</ul>
-		</>
+	const reconnectWallet = async () => {
+		setError(null)
+		await chooseWallet(chosenWallet)
 	}
 
-	const ConnectToWalletStep = () => {
-		if(error) {
-			return <ErrorConnecting/>
-		}
+	return <div className={`modal connect-wallet ${shown ? 'open' : 'closed'}`}>
+		<div className="modal-body">
+			<div className="close">
+				<div className="x" onClick={onClose}>&#9587;</div>
+			</div>
+			{!chosenWallet.name && <ChooseWalletStep onChooseWallet={chooseWallet} />}
+			{(chosenWallet.name && !active) &&
+				<ConnectToWalletStep
+				wallet={chosenWallet}
+				error={error}onTryAgainClick={reconnectWallet}
+				/>}
+			{(chosenWallet.name && active) && <ConnectedView wallet={chosenWallet} account={account} />}
+			<ChooseAccount
+				wallet={chosenWallet}
+				availableAccounts={availableAccounts}
+				active={active}
+				onAccountSelect={activateProvider}
+			/>
+		</div>
+	</div>
+}
 
-		if(chosenWallet.name == 'Ledger') {
-			return <>
-				<div className="title">Plug In Ledger & Enter Pin</div>
-				<p>Open Ethereum application and make sure Contract Data and Browser Support are enabled.</p>
-				<p>Connecting...</p>
-			</>
-		}
+const ChooseWalletStep = ({ onChooseWallet }) => {
+	return <>
+		<div className="title">Connect to a wallet</div>
+		<p>This wallet will be used to sign transactions on Ethereum.</p>
 
-		return <>
-			<div className="title">Connect to a wallet</div>
-			<p>Connecting to {chosenWallet.name} wallet...</p>
-		</>
+		<ul className='wallets'>
+			{
+				WALLETS.map(wallet => {
+					return <li key={wallet.name} className='wallet-option' onClick={() => onChooseWallet(wallet)}>
+						<img alt="wallet-icon" src={wallet.icon} />
+						{wallet.name}
+					</li>
+				})
+			}
+		</ul>
+	</>
+}
+
+const ConnectToWalletStep = ({ error, wallet, onTryAgainClick }) => {
+	if(error) {
+		return <ErrorConnecting error={error} wallet={wallet} onTryAgainClick={onTryAgainClick} />
 	}
 
-	const ErrorConnecting = () => {
+	if(wallet.name == 'Ledger') {
```

Comment:
Should this be `if(wallet.name.includes('Ledger')` since we now have `Legder Live` and `Ledger Legacy`?